string str
float64 intensity
